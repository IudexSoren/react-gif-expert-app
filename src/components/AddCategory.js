import { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 0) {
      setCategories(catLst => [inputValue, ...catLst]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={ handleSubmit }>
      <label>Buscar categor√≠a</label>
      <input
        type="text"
        placeholder="Buscar"
        value={ inputValue }
        onChange={ handleInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default AddCategory;