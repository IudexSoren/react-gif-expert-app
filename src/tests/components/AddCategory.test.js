import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas de AddCategory', () => {

  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={ setCategories }/>);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={ setCategories }/>);
  });

  test('Debe renderizar correctamente el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe cambiar el valor del inputText', () => {
    const input = wrapper.find('input');
    const value = 'Purrumaw';
    input.simulate('change', { target: { value } });
    const initValue = '';

    expect(value).not.toBe(initValue);
  });

  test('No debe hacer nada el formulario', () => {
    const form = wrapper.find('form');
    form.simulate('submit', { preventDefault(){} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test('Debe llamar al setCategories y limpiar el inputText', () => {
    let input = wrapper.find('input');
    const value = 'Purrumaw';
    input.simulate('change', { target: { value } });

    const form = wrapper.find('form');
    form.simulate('submit', { preventDefault(){} });
    input = wrapper.find('input');

    expect(setCategories).toHaveBeenCalled();
    expect(input.prop('value')).toBe('');
  })


})
