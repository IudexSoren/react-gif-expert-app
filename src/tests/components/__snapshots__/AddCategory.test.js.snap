// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de AddCategory Debe renderizar correctamente el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Buscar categor√≠a
  </label>
  <input
    onChange={[Function]}
    placeholder="Buscar"
    type="text"
    value=""
  />
</form>
`;
